### PR TITLE
Refactor SystemDescriptionStore error handling

### PR DESCRIPTION
**Pull request by Werni**

### CODE REVIEW

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
( Success: Merged 14f0b2644fc083aad1094d37c7bccf221519eb20 into 99cd2f3abde658ee652bb2c39a4f311440cc5801 )

File: lib/system_description_store.rb [hunk 1]
```diff
@@ -159,4 +130,20 @@ def get_backup_name(description_name)
 
     backup_name
   end
+
+  def system_description_exists!(description_name)
```
Comment (cornelius) [inline lib/system_description_store.rb]: To be consistent with other methods which do checks the method should probably rather named `validate_existence_of_description`

File: lib/system_description_store.rb [hunk 2]
```diff
@@ -159,4 +130,20 @@ def get_backup_name(description_name)
...
+    if !list.include?(description_name)
+      raise Machinery::Errors::SystemDescriptionNotFound.new(
+        "System description \"#{description_name}\" does not exist."
+      )
+    end
+  end
+
+  def system_description_exists?(description_name)
```
Comment (cornelius) [inline lib/system_description_store.rb]: To be consistent with other methods which do checks the method should probably rather named validate_nonexistence_of_description

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
( Success: Merged 14f0b2644fc083aad1094d37c7bccf221519eb20 into b82d71038ea6d59833d883e4240e71bda4e2c30b )

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
( Success: Merged bc45340378e19205885225aa69fec61449188957 into b82d71038ea6d59833d883e4240e71bda4e2c30b )

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
( Success: Merged bc45340378e19205885225aa69fec61449188957 into 16a2e65250a1b037a429f24bf1408e116a154ad3 )

**Comment by cornelius** [comment]
lgtm

**Pull request merged**